src/sentry/llm/providers/vertex.py

```diff
@@ -0,0 +1,69 @@
+import logging
+from typing import Any
+
+import google.auth
+import google.auth.transport.requests
+import requests
+
+from sentry.llm.providers.base import LlmModelBase
+
+logger = logging.getLogger(__name__)
+
+
+class VertexProvider(LlmModelBase):
```

Comment:
will note that this is for "Palm 2" class models. Google, in classic fashion, has changed the API format for "Gemeni" class models, we'll need to write another provider. will rename this one.

(We aren't using the GCP AI module because it seemed to be quite a heavy one)